### PR TITLE
Implementing T7 status and crispr pair/group input for EP II plate

### PR DESCRIPTION
**Pull request by Jana-A**

### CODE REVIEW

File: root/site/user/report/generic_report_grid.tt [hunk 1]
```diff
@@ -126,12 +126,83 @@
...
+            var override_t7_score = Ext.create( 'Ext.form.ComboBox', {
+                queryMode: 'local',
+                store: t7_percent,
+                displayField: 't_p',
+                forceSelection: true,
+                listeners: {
+                    change: function (field, newValue, oldValue) {
+                        var record = grid.getSelectionModel().getSelection()[0];
+                        console.log(record.data);
+                        $.getJSON( "[% c.uri_for('/api/well/save_well_t7_change') %]", {well_id:record.data['Well ID'], t7_type:'t7_score', t7_value:newValue}, function(data){record.commit()})
```
Comment (jmrein) [inline root/site/user/report/generic_report_grid.tt]: Semantically, nothing that writes data to a REST API should be on a GET request - POST or PUT.
Can use $.post(...) instead (and a corresponding change in the API).

File: ddl/versions/146/up.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+);
+
+CREATE TABLE well_t7 (
+  well_id INT PRIMARY KEY REFERENCES wells(id),
+  t7_score INT,
+  t7_status TEXT,
+  created_by_id INT NOT NULL REFERENCES users(id),
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
+);
+
```
Comment (jmrein) [inline ddl/versions/146/up.sql]: No schema_version bump

File: lib/LIMS2/Report/EPPipelineIIPlate.pm [hunk 3]
```diff
@@ -113,12 +123,29 @@ sub get_well_design {
...
     my $res = $self->model->schema->resultset( 'ProcessCrispr' )->find({ process_id =>  $well_process }, { columns => [ qw/crispr_id/ ] });
-    my $crispr_id = $res->get_column('crispr_id');
+    my $res_pair = $self->model->schema->resultset( 'ProcessCrisprPair' )->find({ process_id =>  $well_process }, { columns => [ qw/crispr_pair_id/ ] });
+    my $res_group = $self->model->schema->resultset( 'ProcessCrisprGroup' )->find({ process_id =>  $well_process }, { columns => [ qw/crispr_group_id/ ] });
+
+    if ($res) {
+        $crispr_info->{crispr_id} = $res->get_column('crispr_id');
+    } elsif ($res_pair) {
+        my $crispr_pair_id = $res_pair->get_column('crispr_pair_id');
+#        my $rec = $self->model->schema->resultset( 'CrisprPair' )->find({ id =>  $crispr_pair_id }, { columns => [ qw/left_crispr_id right_crispr_id/ ] });
```
[the remark this answers is not in the document]
Comment (Jana-A) [reply lib/LIMS2/Report/EPPipelineIIPlate.pm]: yep I've been meaning to take that out thanks.

File: lib/LIMS2/WebApp/Controller/API/PlateWell.pm [hunk 4]
```diff
@@ -170,6 +170,31 @@ sub well_accepted_override_PUT {
...
+    my ( $self, $c ) = @_;
+
+    $c->assert_user_roles( 'edit' );
+
+    my $t7_type = $c->request->param('t7_type');
+    my $well_id = $c->request->param('well_id');
+    my $t7_value = $c->request->param('t7_value');
+
+    print "here $t7_type and $well_id " . $c->user->name . "\n";
+    ##TODO validations
```
[the remark this answers is not in the document]
Comment (Jana-A) [reply lib/LIMS2/WebApp/Controller/API/PlateWell.pm]: ditto

File: lib/LIMS2/WebApp/Controller/API/PlateWell.pm [hunk 5]
```diff
@@ -170,6 +170,31 @@ sub well_accepted_override_PUT {
...
+
+    my $update_well_t7 = $c->model('Golgi')->txn_do(
+        sub {
+            shift->update_well_t7_info( $c->user->name, $c->request->params );
+        }
+    );
+
+    return;
+#    return $self->status_ok( $c, entity => $override );
```
[the remark this answers is not in the document]
Comment (Jana-A) [reply lib/LIMS2/WebApp/Controller/API/PlateWell.pm]: ditto

File: root/site/user/report/generic_report_grid.tt (hunk 1 above)
Comment (Jana-A) [reply root/site/user/report/generic_report_grid.tt]: I agree!

**Pull request merged**